### PR TITLE
Improve dashboard editor restriction settings

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

- Add message to show that only owner can update org settings
- Better error messages when user select an unsupported editor (via url search param `editor`)
- Fix repo-level editor restriction allows to block all editors bug

‼️ not sure about first point, as all settings on org-level only allow `owner` to update it
image

## Related Issue(s)

Fixes EXP-1671, EXP-1672, EXP-1674

## How to test

https://hw-restric6410a20100.preview.gitpod-dev.com/workspaces

Check points on PR description

Join an org as a member https://hw-restric6410a20100.preview.gitpod-dev.com/orgs/join?inviteId=46fa8597-ed0a-4997-8260-068c736121b9

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by laushinka** [comment]
@mustard-mh Could you invite me to your org so I can test as member?

**Comment by mustard-mh** [comment]
@laushinka https://hw-restric6410a20100.preview.gitpod-dev.com/orgs/join?inviteId=46fa8597-ed0a-4997-8260-068c736121b9

**Comment by laushinka** [comment]
Screenshot 2024-03-22 at 10 45 47

I got this even though there were available editors set in the org.

**Comment by mustard-mh** [comment]
> reply to https://github.com/gitpod-io/gitpod/pull/19569#issuecomment-2014721510

@laushinka could you refresh? I just updated org settings

**Comment by laushinka** [comment]
> @laushinka could you refresh? I just updated org settings

Now it works. But I wonder what caused that state, even though there were already two editors available set in the org. 🤔

**Comment by mustard-mh** [comment]
mustard-mh/client is an repo, that only allows few editors. Once those allowed editors is blocked on org-level, there's no editors can be choose. It's known and discussed in our teaming [[internal chat](https://gitpod.slack.com/archives/C05H5UQBW6Q/p1709741929763189?thread_ts=1709740879.899289&cid=C05H5UQBW6Q)]

**Review by laushinka — APPROVED** [review]
Thanks for the quick fixes, Huiwen!

**Comment by mustard-mh** [comment]
> Add message to show that only owner can update org settings
> ‼️ not sure about first point, as all settings on org-level only allow owner to update it

I'm not sure about this point, would be great if @laushinka  or @corygrunk could give some feedback. Not it's not going to block this PR, we can update UI anytime

**Comment by laushinka** [comment]
> > Add message to show that only owner can update org settings
> > ‼️ not sure about first point, as all settings on org-level only allow owner to update it
> 
> I'm not sure about this point, would be great if @laushinka or @corygrunk could give some feedback. But it's not going to block this PR, we can update UI as follow-up

I see the concern about the repetition, although [the feedback about the lack of information](https://gitpod.slack.com/archives/C042X6P86BZ/p1711096643826489?thread_ts=1711091267.539149&cid=C042X6P86BZ) was valid. I'll leave it to @corygrunk re how to surface it better. Yes we can address the UI/UX issues in a follow up.

**Pull request merged**
